File: CSharpCodeBase/utils/bhlib/conditional.lua
local Class = require 'utils.hump.class'

-- Conditionals
local Conditional = Class({init = function(self, func)
    self.func = func
    self.returnCode = B_FAILURE
end})

function Conditional:Behave(creature)
  self.returnCode = self.func(creature) and B_SUCCESS or B_FAILURE
  return self.returnCode
end

function Conditional:Bool(creature)
  return self.func(creature)
end

return Conditional

File: CSharpCodeBase/utils/bhlib/conditional.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class conditional {
 // Conditionals
 public void init(self, func);
     this.func = func;
     this.returnCode = B_FAILURE;
 }})
 public void Behave(creature){
   this.returnCode = this.func(creature)  &&  B_SUCCESS  ||  B_FAILURE;
   return this.returnCode;
 }
 public void Bool(creature){
   return this.func(creature);
 }
 }}